src/lib/builders/table-of-contents/create.ts

```diff
@@ -373,7 +374,7 @@ export function createTableOfContents(args: CreateTableOfContentsArgs) {
 
 					// Add items hash to URL
 					if (id) {
-						history.pushState({}, '', `#${id}`);
+						pushState(`#${id}`, {});
```

Comment:
We avoid using Svelte**Kit**-specific APIs to support non-kit projects.

Perhaps we keep the default at `history.pushState({}, '', `#${id}`)` and expose a prop that receives the `id` as an arg that the user could override if they wanted to.

Something like:

```ts
const defaults = {
	// ...
	pushState: (id) => {
			history.pushState({}, '', `#${id}`
	}
}
```

and then in the click handler:

```ts

if (id) {
	options.pushState.get()(id)
}
```

Or something along those lines. Wdyt @TGlide? Prop name subject to bikeshedding of course (can't think of a nice name atm), but the overall idea.

Reply:
We should provide a default option as @huntabyte suggested, however I would argue that we should mark that option as required. If it's optional, no one is going to provide the svelte-kit specific pushState which is more correct and also prevents the issue for back and forward navigation. 

I assume most people are using melt with svelte-kit, so marking the option as optional, will guarantee that most people will still encounter #1062. 

Of course marking the option as required seems undesired, however it's a difficult challenge to build a library that is mostly used by svelte-kit users while the library tries to not use svelte-kit specific API's which then breaks the apps of most users using this library.

Reply:
I don't think this is good idea. Unless the intention is to become a SK only library, we shouldn't be using/importing any SK specific dependencies, even if we mark them as optional, and _especially_ if they're Vite virtual modules. The bundler will certainly throw a fit when it runs into `await import('$app/navigation')` when it's used in vanilla Vite project and will probably require extra plumbing to not error out for non-SK users.

Reply:
Well this sucks.

Would it be possible to modify the url, and navigate to the element, instead of using pushState?

Reply:
> Would it be possible to modify the url, and navigate to the element, instead of using pushState?

@TGlide I am pretty sure doing that was breaking the smooth scroll https://github.com/melt-ui/melt-ui/pull/860 if using `window.location.href = ...` instead of `pushState`

Reply:
You're thinking putting this into a `runAfterTransition` @TGlide?

Reply:
@huntabyte exactly